### PR TITLE
Add coupon support for monthly subs

### PR DESCRIPTION
**Pull request by differentmatt**
Excluding regions with custom pricing
Removing parent email code

### CODE REVIEW

File: app/schemas/models/product.schema.coffee [hunk 1]
```diff
@@ -18,6 +18,7 @@ module.exports = ProductSchema = {
         properties: {
           code: { type: 'string' }
           amount: { type: 'integer', description: 'Adjusted cost in cents' }
+          duration: { type: 'string'}
```
Comment (sderickson) [inline app/schemas/models/product.schema.coffee]: What is the format, effect of the duration?

File: app/views/core/SubscribeModal.coffee [hunk 2]
```diff
@@ -17,34 +17,47 @@ module.exports = class SubscribeModal extends ModalView
...
-    @subType = utils.getQueryVariable('subtype', me.getSubModalGroup())
 
   onLoaded: ->
     @basicProduct = @products.getBasicSubscriptionForUser(me)
+    # Process basic product coupons unless custom region pricing
+    if @couponID and @basicProduct.get('coupons')? and @basicProduct?.get('name') is 'basic_subscription'
+      @basicCoupon = _.find(@basicProduct.get('coupons'), {code: @couponID})
+      # Always use both-subs UX test group when basic product coupon, and delay identify until we can decide
+      @subType = if utils.getQueryVariable('subtype')?
+        me.getSubModalGroup(utils.getQueryVariable('subtype'))
```
Comment (sderickson) [inline app/views/core/SubscribeModal.coffee]: This seems like this and the other calls in this function should be 'setSubModalGroup`, as its purpose is to set `User.subModalGroup`? I couldn't find any instances of it being used anyway, do we even need that prop?

File: app/views/core/SubscribeModal.coffee [hunk 3]
```diff
@@ -84,7 +90,7 @@ module.exports = class SubscribeModal extends ModalView
 
   afterRender: ->
     super()
-    @setupParentButtonPopover()
+    # TODO: does this work?
```
Comment (sderickson) [inline app/views/core/SubscribeModal.coffee]: Does what work?

Comment (differentmatt) [reply app/views/core/SubscribeModal.coffee]: Not after very limited testing.. ;)

File: app/schemas/models/product.schema.coffee (hunk 1 above)
Comment (differentmatt) [reply app/schemas/models/product.schema.coffee]: Will add a comment.  It's a Stripe duration, and we only support `once` currently.

File: app/views/core/SubscribeModal.coffee (hunk 2 above)
Comment (differentmatt) [reply app/views/core/SubscribeModal.coffee]: We have to force a specific test group in this scenario, and I want the analytics to be correct once we know which group we'll need.

**Pull request closed without merging**